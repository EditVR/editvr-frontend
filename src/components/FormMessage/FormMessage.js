/**
 * @file FormMessage.js
 * Exports a component that renders a given form message.
 */

import React from 'react';
import PropTypes from 'prop-types';
import { Typography } from '@material-ui/core';

const FormMessage = ({ children }) =>
  children ? <Typography variant="subheading">{children}</Typography> : null;

FormMessage.propTypes = {
  children: PropTypes.node
};

FormMessage.defaultProps = {
  children: null
};

export default FormMessage;
