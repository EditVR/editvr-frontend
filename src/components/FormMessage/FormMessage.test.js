/**
 * @file FormMessage.test.js
 * Contains tests for FormMessage.js.
 */

import React from 'react';
import renderer from 'react-test-renderer';

import FormMessage from './FormMessage';

describe('<FormMessage />', () => {
  it('Matches its snapshot', () => {
    expect(
      renderer
        .create(<FormMessage>Error: your input is borked lol</FormMessage>)
        .toJSON()
    ).toMatchSnapshot();
  });
});
