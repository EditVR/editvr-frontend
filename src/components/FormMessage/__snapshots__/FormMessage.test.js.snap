// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormMessage /> Matches its snapshot 1`] = `
<h3
  className="MuiTypography-root-1 MuiTypography-subheading-8"
>
  Error: your input is borked lol
</h3>
`;
